db/migrate/20200524074928_create_service_resource_attachments.rb

```diff
@@ -0,0 +1,13 @@
+class CreateServiceResourceAttachments < ActiveRecord::Migration[5.1]
+  def change
+    create_table :service_resource_attachments do |t|
+      t.references :storage_service, :type => :bigint, :index => true
+      t.references :storage_resource, :type => :bigint, :index => true
+      t.boolean :compliant
+      t.references :ems, :type => :bigint, :index => true, :references => :ext_management_system
```

Comment:
You can use join_table notation to create this if it is a simple has and belongs_to many type table, also then you don't need the ems reference since it is going to be references from the services and resources sides

Reply:
we originally have a compliant field that will be synced from the provider autosde. We will remove the compliance field as it will not be used in the initial release. We will add it in future release so it may be better to keep this table without the automatic join for easy sync with the ems

Reply:
If this is actually something that is actually representing a provider backed object then I'm good with this not being a join table (which typically just represent connections between two other records)